### ISSUE TITLE
Center page

### ISSUE THREAD

**Issue by gigisforza70** [Bhuvaneshw/PdfViewer#7]
In the horizontal view of the document the page is moved to the top, and is not centered. In the vertical view you see one page and half of the other. I think that this is also connected to #1

**Comment by Bhuvaneshw** [comment]
Can you elaborate the issue with vertical view? Note that Vertical view is not like view pager.

**Comment by gigisforza70** [comment]
I would like both horizontal and vertical display like in this app https://github.com/Sav22999/sav-pdf-viewer-pro see screenrecorder

https://github.com/user-attachments/assets/22d3cfcb-d208-4960-8972-bee4f163fbf4

**Comment by Bhuvaneshw** [comment]
Got it

**Comment by Bhuvaneshw** [comment]
Can close the issue if this commit(a5dad08da8eed526c018f6c72250da537cec4f84) satisfies the requirement.

**Comment by gigisforza70** [comment]
Hi, I tested the changes, page snap is fine. The page is not centered yet, let me explain: as you can see from these screenshots  the page display in your app is too high, leaving a lot of space at the bottom. In the other screen instead the page is "centered" that is the space at the top of the page and the space at the bottom of the page are the same

![IMG_20241217_121937_994](https://github.com/user-attachments/assets/235ec4ca-b47f-45c0-8b4d-48ea3145224c)
![IMG_20241217_121938_290](https://github.com/user-attachments/assets/ce83bd06-80e8-47c3-a9b4-1325821512a1)

**Comment by Bhuvaneshw** [comment]
Did you set the alignment mode to center both?

![Screenshot_20241217_170809_PdfViewerDemo](https://github.com/user-attachments/assets/a117477b-1521-403d-8a81-798b2fe23482)

**Comment by gigisforza70** [comment]
Yes, sorry, I forgot that setting. I would say that basically all that is missing is the ability to scroll pages individually, with a view like ViewPager, setting like in the screen recorder, basically scrolling one page at a time.
Then this issue can be closed and also #1

**Comment by Bhuvaneshw** [comment]
I tried that, but the page alignment is not supported to be like view pager. Let me try again once and if I can't do it, then I close this issue!

**Comment by gigisforza70** [comment]
I don't know if it can help you, but there is this app that says it is based on pdf.js which has this default function,
https://github.com/GrapheneOS/PdfViewer
but everything else is missing.
If you tell me the code that controls the page scrolling speed, I can do a more in-depth search and maybe find a solution

**Comment by gigisforza70** [comment]
After some research it seems that using the right and left buttons with zoom set to "page fit" as they are on the GrapheneOS Pdf based on pdf.js the scrolling occurs for a single page. In fact, to change pages in the GrapheneOS Pdf you have to use the right and left keys.
These two comments prove it:

https://github.com/mozilla/pdf.js/issues/1380 
https://github.com/mozilla/pdf.js/issues/3240

The left and right keys were introduced here https://github.com/mozilla/pdf.js/commit/d15847826e19bda52e26cbdfcea57c3373137f2c
I find the use of the left and right keys uncomfortable, I don't know if it's possible to have the same behavior without having to use the keys

Actually there is a PR that was right for us but it has been abandoned maybe it can be of help https://github.com/mozilla/pdf.js/pull/2709

**Comment by Bhuvaneshw** [comment]
> I don't know if it can help you, but there is this app that says it is based on pdf.js which has this default function, https://github.com/GrapheneOS/PdfViewer but everything else is missing. If you tell me the code that controls the page scrolling speed, I can do a more in-depth search and maybe find a solution

They are using Pdf.js directly to render page, but I'm using the Pdf Viewer that is bundled into FireFox(https://mozilla.github.io/pdf.js/web/viewer.html).

**Comment by Bhuvaneshw** [comment]
I think this singlePageArrangement satisfies the issue

Try this
<pre>
pdfViewer.pageScrollMode = PdfViewer.PageScrollMode.VERTICAL // or PdfViewer.PageScrollMode.HORIZONTAL
pdfViewer.singlePageArrangement = true
pdfViewer.pageAlignMode = PdfViewer.PageAlignMode.CENTER_BOTH // after setting single page mode to true
</pre>

**Comment by gigisforza70** [comment]
hi, I tested the changes, let's say they solved a different problem that I wanted to open, now with that option active even in portrait mode a single page is displayed correctly, and even on the tablet in landscape mode now I can finally view a single page, before I saw 3, so this change is perfect for viewing a single page in portrait and landscape mode on tablets. However, it does not solve the single scrolling per page, in fact scrolling from one page to another is still too fast and you can't scroll one page at a time using the touch. 

I ran some tests that confirmed my thesis: connecting a bluetooth keyboard to the smartphone and using the "right" and "left" arrow keys in portrait mode and the "up" and "down" arrow keys in landscape mode, scrolling occurs for a single page "without animation". 
Touch currently replicates the scrolling of the mouse wheel movement.

What I was wondering is it possible to replicate the same behavior of the bluetooth keyboard using the touch? Maybe in the touch you could add the scrolling animation, but without animation it would be fine anyway. I'll leave you a screenrecorder using the keyboard.

https://github.com/user-attachments/assets/d842c75d-37ff-469f-9a97-9f80da19af40

**Comment by gigisforza70** [comment]
Yes indeed it is difficult to scroll currently with that option active.

However I think I have found the solution to scroll only one page at a time without "slowing down the swipe". 
I think the solution is this :

goToNextPage() = goToPage(currentPage + 1)
   
goToPreviousPage() = goToPage(currentPage - 1)

it could be called "single page scrolling" and the current one "continuous scrolling"

**Comment by Bhuvaneshw** [comment]
I think the scrolling is ok now!

**Comment by gigisforza70** [comment]
Ok, now it seems to work fine, although the option is only available if you select "Use compose Pdf Viewer"
Thanks

**Comment by Bhuvaneshw** [comment]
Will be added to xml pdf viewer in next commits
